src/Mod/Sketcher/Gui/TaskSketcherElements.cpp

```diff
@@ -449,176 +750,162 @@ void TaskSketcherElements::onSelectionChanged(const Gui::SelectionChanges& msg)
+                        || item->GeometryType == Part::GeomLineSegment::getClassTypeId()
+                        || item->GeometryType == Part::GeomBSplineCurve::getClassTypeId())) {
+                    item->isEndPointSelected = !item->isEndPointSelected;
+                }
+                else {
+                    item->isLineSelected = !item->isLineSelected;
+                }
+                item->clickedOn = ElementType::none;
+            }
+            else if (multipleconsecutiveselection && previouslySelectedItemIndex >= 0 && !rightClickOnSelected &&
+                ((i > focusItemIndex && i < previouslySelectedItemIndex) || (i<focusItemIndex && i>previouslySelectedItemIndex))) {
+                if (item->GeometryType == Part::GeomPoint::getClassTypeId()) {
+                    item->isStartingPointSelected = true;
+                }
+                else {
+                    item->isLineSelected = true;
                 }
             }
-        }
-
-        // first update the listwidget
-        switch(element){
-          case 0:
-              ite->setSelected(ite->isLineSelected);
-              break;
-          case 1:
-              ite->setSelected(ite->isStartingPointSelected);
-              break;
-          case 2:
-              ite->setSelected(ite->isEndPointSelected);
-              break;
-          case 3:
-              ite->setSelected(ite->isMidPointSelected);
-              break;
-        }
 
-        // now the scene
-        std::stringstream ss;
-        int vertex;
+            // first update the listwidget. Item is selected if at least one element of the geo is selected.
+            item->setSelected(item->isLineSelected || item->isStartingPointSelected || item->isEndPointSelected || item->isMidPointSelected);
 
-        if (ite->isLineSelected) {
-            ss << "Edge" << ite->ElementNbr + 1;
-            elementSubNames.push_back(ss.str());
-        }
+            // now the scene
+            std::stringstream ss;
+            int vertex;
 
-        if (ite->isStartingPointSelected) {
-            ss.str(std::string());
-            vertex= ite->StartingVertex;
-            if (vertex!=-1) {
-                ss << "Vertex" << vertex + 1;
+            if (item->isLineSelected) {
+                ss << "Edge" << item->ElementNbr + 1;
                 elementSubNames.push_back(ss.str());
             }
-        }
 
-        if (ite->isEndPointSelected) {
-            ss.str(std::string());
-            vertex= ite->EndVertex;
-            if (vertex!=-1) {
-                ss << "Vertex" << vertex + 1;
-                elementSubNames.push_back(ss.str());
+            if (item->isStartingPointSelected) {
+                ss.str(std::string());
+                vertex = item->StartingVertex;
+                if (vertex != -1) {
+                    ss << "Vertex" << vertex + 1;
+                    elementSubNames.push_back(ss.str());

```

Comment:
I'm not sure what is your suggestion here?

Reply:
I think this is overdoing it no?

Reply:
Could be :)

Reply:
It has an easy refactor with a lambda. I will be doing it.

Reply:
Maybe @PaddleStroke can do it as part of learning curve ? :)

Reply:
@PaddleStroke 

Do you want to do it yourself or prefer me to do it?

Reply:
Please go ahead while you're at it. I don't know exactly what is a lambda, I'll learn from your commit.

Reply:
Ok. I just pushed the refactor.

A lambda can be seen as an object that is local function. It is a very powerful tool in many situations. This is just one situation.

So before lambdas, if wanting to refactor, we would need to create a function (either a local member function, or a free function). This would be located elsewhere and it would be difficult to keep the context when reading the function.

Here the refactor is just realising we are repeating a pattern, identifying the parameters, making the common part into a lambda, and calling the lambda as many times as the pattern repeated itself.

As a further advantage, a local lambda can use captures (what is inside the brackets), which basically allows you to pass by reference (or value), some local objects that are not necessary parameters of the function, but are necessary to run it.